meteor/client/ui/Shelf/MultiViewPanel.tsx

```diff
@@ -0,0 +1,189 @@
+	unfinishedPieces: {
+		[key: string]: PieceInstance[]
+	}
+	nextPieces: {
+		[key: string]: PieceInstance[]
+	}
+}
+
+export class MultiViewPanelInner extends MeteorReactComponent<Translated<IAdLibPanelProps & IMultiViewPanelProps & IAdLibPanelTrackedProps & IMultiViewPanelTrackedProps>, IState> {
+
+	constructor (props: Translated<IAdLibPanelProps & IAdLibPanelTrackedProps>) {
+		super(props)
+
+	}
+
+	isAdLibOnAir (adLib: AdLibPieceUi) {
+		if (this.props.unfinishedPieces[unprotectString(adLib._id)] && this.props.unfinishedPieces[unprotectString(adLib._id)].length > 0) {
+			return true
+		}
+		return false
+	}
+
+	isAdLibNext (adLib: AdLibPieceUi) {
+		if (this.props.nextPieces[unprotectString(adLib._id)] && this.props.nextPieces[unprotectString(adLib._id)].length > 0) {
+			return true
+		}
+		return false
+	}
+
+	onToggleSticky = (sourceLayerId: string, e: any) => {
+		if (this.props.playlist && this.props.playlist.currentPartInstanceId && this.props.playlist.active) {
+			const { t } = this.props
+			doUserAction(t, e, 'Start Sticky Piece', (e) => MeteorCall.userAction.sourceLayerStickyPieceStart(e,
+				this.props.playlist._id, sourceLayerId
+			))
+		}
+	}
+
+	toggleAdLib (e: any, piece?: AdLibPieceUi, queueWhenOnAir?: boolean) {
+		const { t } = this.props
+		if (!piece) {
+			return
+		}
+
+		if (piece.invalid) {
+			NotificationCenter.push(new Notification(
+				t('Invalid AdLib'),
+				NoticeLevel.WARNING,
+				t('Cannot play this AdLib because it is marked as Invalid'),
+				'toggleAdLib'))
+			return
+		}
+
+		const currentPartInstanceId = this.props.playlist.currentPartInstanceId
+
+		if ((!this.isAdLibOnAir(piece) || queueWhenOnAir) && this.props.playlist && currentPartInstanceId) {
+			if (!piece.isGlobal) {
+				doUserAction(t, e, 'Start Adlib', (e) => MeteorCall.userAction.segmentAdLibPieceStart(e,
+					this.props.playlist._id, currentPartInstanceId, piece._id, true
+				))
+			} else if (piece.isGlobal && !piece.isSticky) {
+				doUserAction(t, e, 'Start global Adlib', (e) => MeteorCall.userAction.baselineAdLibPieceStart(e,
+					this.props.playlist._id, currentPartInstanceId, piece._id, true
+				))
+			} else if (piece.isSticky) {
+				this.onToggleSticky(piece.sourceLayerId, e)
+			}
+		}
+	}
+
+	take = (e: any) => {
+		const { t } = this.props
+		if (this.props.studioMode) {
+			doUserAction(t, e, 'Take', (e) => MeteorCall.userAction.take(e, this.props.playlist._id))
+		}
+	}
+
+	onAction = (e: any, piece?: AdLibPieceUi) => {
+		switch (this.props.panel.role) {
+			case RundownLayoutMultiViewRole.QUEUE:
+				this.toggleAdLib(e, piece, true)
+				break
+			case RundownLayoutMultiViewRole.TAKE:
+				this.take(e)
+				break
+			case RundownLayoutMultiViewRole.PROGRAM:
+				break
+		}
+	}
+
+	render () {
+		const isTake = this.props.panel.role === RundownLayoutMultiViewRole.TAKE
+		const isProgram = this.props.panel.role === RundownLayoutMultiViewRole.PROGRAM
+		const isLarge = isProgram || isTake
```

Comment:
I don't like this isLarge behaviour being done based on the action. This should be made into a separate config field

Reply:
I have moved this to the setting "Place label below panel" tied to the "labelBelowPanel" property https://github.com/olzzon/tv-automation-server-core/commit/8ac362f4db862137d5be13c0e9f89270b442221f